src/context/devices.context.tsx

```diff
@@ -0,0 +1,254 @@
+import { createContext, useCallback, useEffect,useState } from "react";
+import { App, Cloud ,ID, Value, Device, Meta } from "waziup";
+import { Devices,getNetworkDevices } from "../utils/systemapi";
+import GlobalDialog from "../components/shared/GlobalDialog";
+export type SensorX =  {
+    id: ID;
+    name: string;
+    value: Value;
+    modified: Date;
+    created: Date;
+    time: Date;
+    meta: Meta;
+    kind: string
+    quantity: string
+    unit: string
+}
+export type ActuatorX =  {
+    id: ID;
+    name: string;
+    value: Value;
+    modified: Date;
+    created: Date;
+    time: Date;
+    meta: Meta;
+    kind: string
+    quantity: string
+    unit: string
+}
+type DeviceX = Device & {
+    sensors: SensorX[]
+}
+interface ContextValues{
```

Comment:
There is one big Context with everything inside. It is better to have several smaller contexts: devices, apps, profile...